contracts/src/Gateway.sol

```diff
@@ -206,7 +206,7 @@ contract Gateway is IGateway, IInitializable {
         // Calculate a gas refund, capped to protect against huge spikes in `tx.gasprice`
         // that could drain funds unnecessarily. During these spikes, relayers should back off.
         uint256 gasUsed = _transactionBaseGas() + (startGas - gasleft());
-        uint256 refund = gasUsed * Math.min(tx.gasprice, message.maxFeePerGas);
+        uint256 refund = Math.min(maxDispatchGas * message.maxFeePerGas, gasUsed * tx.gasprice);
```

Comment:
`message.maxDispatchGas` is what we set in BH for estimating the gas cost and charge from the user, assuming it's somehow underestimated and far less than real `gasUsed` here, we may refund too much, so IMO better to use that to calculate the minimal refund.

Reply:
Lets rather consider this separately. It is unrelated to refunding surplus weight on AssetHub.

Reply:
Reverted in https://github.com/Snowfork/snowbridge/pull/1096/commits/f426fcc931fa23d330383d988d800f5c1a447c35

@vgeddes Do you think the fix here makes sense? If so I'll add another PR for it.